cmd/flux/cluster_info.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1"
+)
+
+// bootstrapLabels are labels put on a resource by kustomize-controller. These labels on the CRD indicates
+// that flux has been bootstrapped.
+var bootstrapLabels = []string{
+	fmt.Sprintf("%s/name", kustomizev1.GroupVersion.Group),
+	fmt.Sprintf("%s/namespace", kustomizev1.GroupVersion.Group),
+}
+
+// fluxClusterInfo contains information about an existing flux installation on a cluster.
+type fluxClusterInfo struct {
+	// bootstrapped indicates that Flux was installed using the `flux bootstrap` command.
+	bootstrapped bool
+	// managedBy is the name of the manager who installed Flux.
+	managedBy string
+	// version is the Flux version number in semver format.
+	version string
+}
+
+// getFluxClusterInfo checks the GitRepository CRD for labels passed into the function. It returns the version of Flux
+// (from the app.kubernetes.io/version label), a boolean to indicate whether the CRD was found
+// with the required labels and an error if the CRD query failed.
```

Comment:
```suggestion
// getFluxClusterInfo returns information on the Flux installation running on the cluster. If the information cannot be retrieved, the boolean return value will be false. If an error occurred, the returned error will be non-nil.
// This function retrieves the GitRepository CRD from the cluster and checks it for a set of labels used to determine the Flux version and how Flux was installed.
```
Not sure about line length here, GitHub's UI doesn't help a lot. You might have to re-wrap it.